Sources/MapboxDirections/MBDirections.swift

```diff
@@ -129,143 +137,238 @@ open class Directions: NSObject {
         } else {
             apiEndpoint = URL(string:(defaultApiEndPointURLString ?? "https://api.mapbox.com"))!
         }
-
+        
     }
-
+    
     /**
      Initializes a newly created directions object with an optional access token.
-
+     
      The directions object sends requests to the [Mapbox Directions API](https://docs.mapbox.com/api/navigation/#directions) endpoint.
-
+     
      - parameter accessToken: A Mapbox [access token](https://docs.mapbox.com/help/glossary/access-token/). If an access token is not specified when initializing the directions object, it should be specified in the `MGLMapboxAccessToken` key in the main application bundle’s Info.plist.
      */
-    @objc public convenience init(accessToken: String?) {
+    public convenience init(accessToken: String?) {
         self.init(accessToken: accessToken, host: nil)
     }
-
+    
     // MARK: Getting Directions
-
+    
+    
     /**
      Begins asynchronously calculating routes using the given options and delivers the results to a closure.
-
+     
      This method retrieves the routes asynchronously from the [Mapbox Directions API](https://www.mapbox.com/api-documentation/navigation/#directions) over a network connection. If a connection error or server error occurs, details about the error are passed into the given completion handler in lieu of the routes.
-
+     
      Routes may be displayed atop a [Mapbox map](https://www.mapbox.com/maps/).
-
+     
      - parameter options: A `RouteOptions` object specifying the requirements for the resulting routes.
      - parameter completionHandler: The closure (block) to call with the resulting routes. This closure is executed on the application’s main thread.
      - returns: The data task used to perform the HTTP request. If, while waiting for the completion handler to execute, you no longer want the resulting routes, cancel this task.
      */
-    @objc(calculateDirectionsWithOptions:completionHandler:)
+    
+    
     @discardableResult open func calculate(_ options: RouteOptions, completionHandler: @escaping RouteCompletionHandler) -> URLSessionDataTask {
-        let fetchStartDate = Date()
-        let task = dataTask(forCalculating: options, completionHandler: { (json) in
+        let completion = completionHandler as MapboxRouteCompletionHandler
+        let fetchStart = Date()
+        let request = urlRequest(forCalculating: options)
+        let requestTask = URLSession.shared.dataTask(with: request) { (possibleData, possibleResponse, possibleError) in
             let responseEndDate = Date()
-            let response = options.response(from: json)
-            if let routes = response.1 {
-                self.postprocess(routes, fetchStartDate: fetchStartDate, responseEndDate: responseEndDate, uuid: json["uuid"] as? String)
+            guard let response = possibleResponse, ["application/json", "text/html"].contains(response.mimeType) else {
```

Comment:
Why does this method accept HTML-formatted responses while the other `calculate` methods only accept JSON?